test/mocha/configuration-parser.spec.ts

```
@@ -0,0 +1,85 @@
+import { expect } from 'chai'
+import { configurationParser } from '../../src/mocha/configuration-parser'
+
+describe('mocha configuration-parser', function () {
+  it('supports all basic options', function () {
+    const rawConfig = [
+      '--bail',
+      '# I am a comment',
+      '--grep foo',
+      '--slow 12',
+      '--timeout 1000',
+      '--retries 42'
+    ].join('\n')
+    const parsed = configurationParser(rawConfig)
+    expect(parsed).to.deep.equal({
+      bail: true,
+      grep: 'foo',
+      retries: '42',
+      slow: '12',
+      timeout: '1000'
+    })
+  })
+  it('throws when there are require flags', function () {
+    const rawConfig = [
+      '--bail',
+      '# I am a comment',
+      '--grep foo',
+      '--slow 12',
+      '--timeout 1000',
+      '--require ./foo.js',
+      '--retries 42'
+    ].join('\n')
+    expect(() => configurationParser(rawConfig)).to.throw(
```

I wouldn't expect an error in this case.
The reason is this:
We don't want someone to change is mocha.opts file and remove this from there.
What I would expect is like this:
if require is specified in mocha.opts we should check if the file appears in the files property in the config in bit.json. if no we should throw an error which tells the user to add it there.
We need to define it as file since we need bit to store it and create it in the consumer environment.
We don't need the user to add it to the bit.json rawConfig.require or rawConfig.filesRequire (this we will do internally by the parser).